cmd/sink/filters/filters.go

```diff
@@ -128,34 +127,83 @@ func (f *Filters) createFilters(
 	if err != nil {
 		return archiveMap, deleteMap, archiveOnDeleteMap, err
 	}
+
 	var errList []error
 	for _, resource := range resources {
 		gvk := schema.FromAPIVersionAndKind(resource.Selector.APIVersion, resource.Selector.Kind)
 		namespaceGvk := NamespaceGVKFromNamespaceAndGvk(namespace, gvk)
 		if resource.ArchiveWhen != "" {
```

Comment:
The three ifs starting here are all identical code. Why not split out these into a local method that can be called for each?

Reply:
I was trying to not change too much, but done.